awscli/arguments.py

```diff
@@ -440,11 +438,11 @@ def add_to_params(self, parameters, value):
             parameters[self._serialized_name] = unpacked
 
     def _unpack_argument(self, value):
-        service_name = self.operation_object.service.endpoint_prefix
-        operation_name = xform_name(self.operation_object.name, '-')
+        service_name = self._operation_model.service_model.endpoint_prefix
```

Comment:
That is fine with me. I agree minimalistic changes for this PR is ideal. We can fix it after it gets merged since it will require a botocore PR as well.